src/private_broadcast.h

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) 2023-present The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or https://opensource.org/license/mit/.
+
+#ifndef BITCOIN_PRIVATE_BROADCAST_H
+#define BITCOIN_PRIVATE_BROADCAST_H
+
+#include <primitives/transaction.h>
+#include <sync.h>
+#include <threadsafety.h>
+#include <util/hasher.h>
+
+#include <chrono>
+#include <map>
+#include <unordered_map>
+
+/**
+ * Store a list of transactions to be broadcast privately. Supports the following operations:
```

Comment:
cade8e833197a202dbb03c50359396ecd437ac8c

optional, could use a comment explaining the priority metric. Looks like transactions with fewer broadcasts or older broadcasts are prioritized. A unit test would help make that clear as well.